### ISSUE TITLE
Handle HTTP 204 (No Content) Responses

### ISSUE THREAD

**Issue by abrahamy** [zhengcan/apisdk-rs#5]
Requests to endpoints that return a HTTP 204 status code without any response body end up failing with the following error: `Decode response error: application/json => error decoding response body`.

## Sample Server
```python
# ./example.py
# pip install "fastapi[standard]"
# fastapi dev example.py
from fastapi import FastAPI, status
from pydantic import BaseModel

app = FastAPI()


class Payload(BaseModel):
    name: str


@app.post("/test", status_code=status.HTTP_204_NO_CONTENT)
def reply_204(data: Payload):
    print("No return statement here")
```

## Sample Client
```rust
use apisdk::serde_json::json;
use apisdk::{http_api, send_json, ApiResult};

#[http_api("https://localhost:8000/")]
pub struct SampleAPI;

impl SampleAPI {
    pub async fn reply_204(&self) -> ApiResult<()> {
        let payload = json!({"name": "test"});
        let req = self.post("/test").await?;
        send_json!(req, payload, ()).await
    }
}
```

Calling the `reply_204` method of `SampleAPI` should return an `Ok(())` but it always returns an error even when the sample server responds correctly with an HTTP 204 code.

**Comment by zhengcan** [comment]
```
curl -v localhost:8000/test -d '{"name":"test"}' -H "Content-Type: application/json"
* Host localhost:8000 was resolved.
*   Trying 127.0.0.1:8000...
* Connected to localhost (127.0.0.1) port 8000
> POST /test HTTP/1.1
> Host: localhost:8000
> User-Agent: curl/8.7.1
> Accept: */*
> Content-Type: application/json
> Content-Length: 15
> 
* upload completely sent off: 15 bytes
< HTTP/1.1 204 No Content
< date: Mon, 18 Nov 2024 15:08:08 GMT
< server: uvicorn
< content-type: application/json
< 
* Connection #0 to host localhost left intact
```

Currently, the library will try to parse the response as JSON according the `content-type` header.

After a little research of 204 No Content, I think we should ignore all payload.
- https://github.com/mdn/content/issues/36146
- https://github.com/httpwg/http-core/issues/26
 
So, let me patch the code :D

**Comment by zhengcan** [comment]
I've published a new version (v0.0.16) to fix this issue
- https://crates.io/crates/apisdk/0.0.16
- commit: https://github.com/zhengcan/apisdk-rs/commit/8e7b8ce6fe9b25afbec78e9a8f87f94b4a48f068

**Comment by abrahamy** [comment]
Thanks for taking the time to fix this so quickly🙏
